Studies/source/VariableDistribution.cxx

```diff
@@ -0,0 +1,226 @@
+    {
+        static const std::set<std::string> EnabledBranches_read = {
+            "eventEnergyScale", "q_1", "q_2", "jets_p4", "extraelec_veto", "extramuon_veto ", "SVfit_p4",
+            "pfMET_p4", "p4_1", "p4_2"
+        };
+        return EnabledBranches_read;
+    }
+
+    static SampleEntryCollection ReadConfig(const std::string& cfg_file){
+        std::ifstream f(cfg_file);
+        SampleEntryCollection collection;
+        while(f.good()){
+            std::string line;
+            std::getline(f, line);
+            if (line.size()==0 || line.at(0)=='#')
+                continue;
+            std::istringstream s(line);
+            SampleEntry entry;
+            s>>entry;
+            collection.push_back(entry);
+        }
+        return collection;
+    }
+
+    MvaClassification(const Arguments& _args): args(_args), samples(ReadConfig(args.cfg_file())),
+        outfile(root_ext::CreateRootFile(args.output_file())), split_training_testing(false),
+        vars(split_training_testing,UINT_FAST32_MAX, enabled_vars, disabled_vars)
+    {
+    }
+
+    void Run()
+    {
+
+        auto start_tot = clock::now();
+        MassVar sample_vars;
+
+        auto start = clock::now();
+        for(const SampleEntry& entry:samples)
+        {
+            if ( args.tree_name() == "muTau" && entry.filename == "TT_ext3_eTau.root")  continue;
+            if ( args.tree_name() == "eTau" && entry.filename == "TT_ext3_muTau.root") continue;
+            auto input_file = root_ext::OpenRootFile(args.input_path()+"/"+entry.filename);
+            EventTuple tuple(args.tree_name(), input_file.get(), true, {} , GetEnabledBranches());
+            Long64_t current_entry = 0, tot_entries = 0;
+
+            while(current_entry < tuple.GetEntries()) {
+                tuple.GetEntry(current_entry);
+                const Event& event = tuple.data();
+                if (event.eventEnergyScale != 0 || (event.q_1+event.q_2) != 0 || event.jets_p4.size() < 2
+                    || event.extraelec_veto == true || event.extramuon_veto == true || event.jets_p4[0].eta() > 2.4
+                    || event.jets_p4[1].eta() > 2.4){
+                    current_entry++;
+                    continue;
+                }
+                LorentzVectorE_Float bb = event.jets_p4[0] + event.jets_p4[1];
+                double ellipse_cut = pow(event.SVfit_p4.mass()-116,2)/pow(35.,2) + pow(bb.mass()-111,2)/pow(45.,2);
+                if (ellipse_cut>1){
+                    current_entry++;
+                    continue;
+                }
+                vars.AddEvent(event, entry.mass, entry.weight);
+                current_entry++;
+                tot_entries++;
+            }
+            std::cout << entry << " number of events: " << tot_entries << std::endl;
+        }
+        auto stop = clock::now();
+        std::cout<<"secondi: "<<std::chrono::duration_cast<std::chrono::seconds>(stop - start).count()<<std::endl;
```

Comment:
it is used many times: declare start as the member and define method ReportTiming(bool total)